presto-spi/src/main/java/com/facebook/presto/spi/block/BlockBuilder.java

```diff
@@ -62,6 +62,15 @@ default BlockBuilder writeObject(Object value)
     }
 
     /**
+     * Return a writer to the current entry. The caller can operate on the returned caller to incrementally build the object. This is generally more efficient than
+     * building the object elsewhere and call writeObject afterwards because a large chunk of memory could potentially be unnecessarily copied in this process.
+     */
+    default BlockBuilder nextObjectWriter()
```

Comment:
I would name this `createBlockPositionWriter`, since it can only be used to write `Object` positions that are `Block`s. @dain thoughts?